sdk/core/core-lro/samples/v3/javascript/customPoller.js

```diff
@@ -0,0 +1,175 @@
+      handlers.set(s, callback);
+      return () => handlers.delete(s);
+    },
+    serialize: async () => {
+      await statePromise;
+      return JSON.stringify({
+        state,
+      });
+    },
+    submitted: async () => {
+      await statePromise;
+    },
+    pollUntilDone: async (pollOptions) => {
+      resultPromise ??= (async () => {
+        await statePromise;
+        if (!state) {
+          throw new Error("Poller should be initialized but it is not!");
+        }
+        const { abortSignal: inputAbortSignal } = pollOptions || {};
+        // In the future we can use AbortSignal.any() instead
+        function abortListener() {
+          abortController.abort();
+        }
+        const abortSignal = abortController.signal;
+        if (inputAbortSignal?.aborted) {
+          abortController.abort();
+        } else if (!abortSignal.aborted) {
+          inputAbortSignal?.addEventListener("abort", abortListener, { once: true });
+        }
+
+        try {
+          if (!poller.isDone) {
+            await poller.poll({ abortSignal });
+            while (!poller.isDone) {
+              await delay(currentPollIntervalInMs, { abortSignal });
+              await poller.poll({ abortSignal });
+            }
+          }
+        } finally {
+          inputAbortSignal?.removeEventListener("abort", abortListener);
+        }
+        switch (state.status) {
+          case "succeeded":
+            return poller.result;
+          case "canceled":
+            throw new Error(cancelErrMsg);
+          case "failed":
+            throw state.error;
+          case "notStarted":
+          case "running":
+            throw new Error(`Polling completed without succeeding or failing`);
+        }
+      })().finally(() => {
+        resultPromise = undefined;
+      });
+      return resultPromise;
+    },
+    async poll(pollOptions) {
+      await statePromise;
+      if (!state) {
+        throw new Error("Poller should be initialized but it is not!");
+      }
+      switch (state.status) {
+        case "succeeded":
+          return state;
+        case "canceled":
+          throw new Error(cancelErrMsg);
+        case "failed":
+          throw state.error;
+      }
+      await pollOperation({
+        state,
+        options: pollOptions,
+        setDelay: (pollIntervalInMs) => {
+          currentPollIntervalInMs = pollIntervalInMs;
+        },
+      });
+      await handleProgressEvents();
+      switch (state.status) {
+        case "canceled":
+          throw new Error(cancelErrMsg);
+        case "failed":
+          throw state.error;
+      }
+
+      return state;
+    },
+    then(onfulfilled, onrejected) {
+      return poller.pollUntilDone().then(onfulfilled, onrejected);
+    },
+    catch(onrejected) {
+      return poller.pollUntilDone().catch(onrejected);
+    },
+    finally(onfinally) {
+      return poller.pollUntilDone().finally(onfinally);
+    },
+    [Symbol.toStringTag]: "Poller",
```

Comment:
What is this used for ? Just curious

Reply:
Could be helpful in debugging to tag the object with its name.